### PR TITLE
Configurable Waitress Threads; General Threading Cleanup

### PR DESCRIPTION
**Pull request by adamkrellenstein**
We're heavily IO-bound, so the default number of threads for Waitress (4) is low. We should make this configurable, and we should default to a higher value. (Without this, slow requests should hang, which is what we observed!)

* [x] Double-check the spelling and grammar of all strings, code comments, etc.
* [x] Double-check that all code is deterministic that needs to be
* [x] Add tests to cover any new or revised logic
* [ ] Ensure that the test suite passes
* [x] Update the project [release notes](release-notes/)
* [x] Update the project documentation, as appropriate, with a corresponding Pull Request in the [Documentation repository](https://github.com/CounterpartyXCP/Documentation)

### CODE REVIEW

**Comment by adamkrellenstein** [comment]
Maybe causes this https://github.com/CounterpartyXCP/counterparty-core/issues/2489 when the number of threads is too high??

**Comment by adamkrellenstein** [comment]
```
2024-10-20T03:22:16.513+00:00 - [   DEBUG] - API Request - Get Valid Assets By Owner (verbose=true address=1KfVAroDNucPqRxCqww9LTgduviYMToukQ)                                                                      
2024-10-20T03:22:16.517+00:00 - [   DEBUG] - API Request - Get Valid Assets By Owner (verbose=true address=1KfVAroDNucPqRxCqww9LTgduviYMToukQ) - Response 200 - 6ms                                                 
2024-10-20T03:22:16.521+00:00 - [   ERROR] - Exception on /v2/addresses/events [GET]                                                                                                                                
Traceback (most recent call last):                                                                                                                                                                                  
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask/app.py", line 1455, in wsgi_app                                                                                                             
    response = self.full_dispatch_request()                                                                                                                                                                         
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                         
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask/app.py", line 869, in full_dispatch_request                                                                                                 
    rv = self.handle_user_exception(e)                                                                                                                                                                              
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                              
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask/app.py", line 867, in full_dispatch_request                                                                                                 
    rv = self.dispatch_request()                                                                                                                                                                                    
         ^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                    
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask/app.py", line 852, in dispatch_request                                                                                                      
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)                                                                                                                                        
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                        
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask_httpauth.py", line 174, in decorated                                                                                                        
    return self.ensure_sync(f)(*args, **kwargs)                                                                                                                                                                     
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                     
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/counterpartycore/lib/api/api_server.py", line 342, in handle_route                                                                                
    result = inject_details(db, result)                                                                                                                                                                             
             ^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                             
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/counterpartycore/lib/api/util.py", line 700, in inject_details                                                                                    
    result_list = inject_dispensers(db, result_list)                                                                                                                                                                
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/counterpartycore/lib/api/util.py", line 654, in inject_dispensers                                                                                 
    dispenser_info = ledger.get_dispensers_info(db, dispenser_list)                                                                                                                                                 
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                 
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/counterpartycore/lib/ledger.py", line 1438, in get_dispensers_info                                                                                
    cursor = db.cursor()                                                                                                                                                                                            
             ^^^^^^^^^^^                                                                                                                                                                                            
apsw.ConnectionClosedError: The connection has been closed                                                                                                                                                          
2024-10-20T03:22:16.524+00:00 - [ WARNING] - Closing connection due to pool size limit (API DB).                                                                                                                    
2024-10-20T03:22:16.526+00:00 - [ WARNING] - Closing connection due to pool size limit (API DB).                                                                                                                    
2024-10-20T03:22:16.535+00:00 - [   ERROR] - Exception on /v2/addresses/events [GET]                                                                                                                                
Traceback (most recent call last):                                                                                                                                                                                  
  File "/home/adam_krellenstein/venv/lib/python3.11/site-packages/flask/app.py", line 1455, in wsgi_app                                                                                                             
    response = self.full_dispatch_request()                                                                                                                                                                         
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                     
```

File: counterparty-core/counterpartycore/lib/api/wsgi.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def __init__(self, app, args=None):
         self.args = args
         self.timer_db = get_db_connection(config.API_DATABASE, read_only=True, check_wal=False)
         self.server = waitress.server.create_server(
-            self.app, host=config.API_HOST, port=config.API_PORT
+            self.app, host=config.API_HOST, port=config.API_PORT, threads=20
         )
```
Comment (ouziel-slama) [inline counterparty-core/counterpartycore/lib/api/wsgi.py:278]: `threads=config.WAITRESS_THREADS` ?

**Review by ouziel-slama — APPROVED** [review]

**Comment by ouziel-slama** [comment]
PyTest tests are fixed here https://github.com/CounterpartyXCP/counterparty-core/pull/2485

File: counterparty-core/counterpartycore/lib/api/wsgi.py (hunk 1 above)
Comment (adamkrellenstein) [reply counterparty-core/counterpartycore/lib/api/wsgi.py:278]: 🤦‍♀️

**Review by ouziel-slama — APPROVED** [review]
Nice!

**Pull request merged**